google/resource_cloudfunctions_function_test.go

```diff
@@ -0,0 +1,476 @@
+					testAccCloudFunctionsFunctionTrigger(FUNCTION_TRIGGER_BUCKET, &function),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"timeout", "61"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"entry_point", "helloGCS"),
+					resource.TestCheckResourceAttr(funcResourceName,
+						"trigger_bucket", bucketName),
+				),
+			},
+			{
+				ResourceName:      funcResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckCloudFunctionsFunctionDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_cloudfunctions_function" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		project := rs.Primary.Attributes["project"]
+		region := rs.Primary.Attributes["region"]
+		cloudFuncId := &cloudFunctionId{
+			Project: project,
+			Region:  region,
+			Name:    name,
+		}
+		_, err := config.clientCloudFunctions.Projects.Locations.Functions.Get(cloudFuncId.cloudFunctionId()).Do()
+		if err == nil {
+			return fmt.Errorf("CloudFunctions still exists")
+		}
+
+	}
+
+	return nil
+}
+
+func testAccCloudFunctionsFunctionExists(n string, function *cloudfunctions.CloudFunction) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		name := rs.Primary.Attributes["name"]
+		project := rs.Primary.Attributes["project"]
+		region := rs.Primary.Attributes["region"]
+		cloudFuncId := &cloudFunctionId{
+			Project: project,
+			Region:  region,
+			Name:    name,
+		}
+		found, err := config.clientCloudFunctions.Projects.Locations.Functions.Get(cloudFuncId.cloudFunctionId()).Do()
+		if err != nil {
+			return fmt.Errorf("CloudFunctions Function not present")
+		}
+
+		*function = *found
+
+		return nil
+	}
+}
+
+func testAccCloudFunctionsFunctionSource(n string, function *cloudfunctions.CloudFunction) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if n != function.SourceArchiveUrl {
+			return fmt.Errorf("Expected source to be %v, got %v", n, function.EntryPoint)
+		}
+		return nil
+	}
+}
+
+func testAccCloudFunctionsFunctionTrigger(n int, function *cloudfunctions.CloudFunction) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		switch n {
+		case FUNCTION_TRIGGER_HTTP:
+			if function.HttpsTrigger == nil {
+				return fmt.Errorf("Expected trigger_http to be set")
+			}
+		case FUNCTION_TRIGGER_BUCKET:
+			if function.EventTrigger == nil {
+				return fmt.Errorf("Expected trigger_bucket to be set")
+			}
+			if strings.Index(function.EventTrigger.EventType, "cloud.storage") == -1 {
+				return fmt.Errorf("Expected trigger_bucket to be set")
```

Comment:
and this one would be something like "Expected cloud.storage EventType" (or even better, "Expected cloud.storage EventType, found %s")

Reply:
Fixed